### PR TITLE
WIP: Add ci to avoid build failures 

### PR DESCRIPTION
**Pull request by frejsoya**
The existing Travis CI setup appears dead. given the few PR/issues. Verifying builds would be helpful :)

Build output example: https://github.com/frejsoya/ocaml-kafka/actions/runs/6084853534

Using Github actions as that was the easiest to do.

- Builds with 4.14 and 5.00 on mac/ubuntu. 
- Updated to js core 0.16
- Updated ocamlformat. as 0.13 requires Ocaml < 4.11. 

I have avoided including the autoformatting changes with v 0.26.  I have created separate PR  that:
- Bumps to 0.26.
- Includes formatting changes.

I can then rebase this PR on top.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
```
Comment (frejsoya) [inline .github/dependabot.yml:1]: Not sure this is actually needed.

File: .ocamlformat [hunk 2]
```diff
@@ -1,2 +1,2 @@
-version=0.13.0
-indicate-multiline-delimiters=no
```
Comment (frejsoya) [inline .ocamlformat:2]: indicate-multiline-delimiters=no should be the default

**Comment by didier-wenzek** [comment]
Thank you for this PR. I'm a bit busy these days, but I will take the time to have a look before the end of this week.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**